ipn/serve.go

```diff
@@ -355,6 +383,60 @@ func CheckFunnelPort(wantedPort uint16, node *ipnstate.PeerStatus) error {
 	return deny(portsStr)
 }
 
+// ExpandProxyTargetValue expands the supported target values to be proxied
+// allowing for input values to be a port number, a partial URL, or a full URL
+// including a path.
+//
+// examples:
+//   - 3000
+//   - localhost:3000
+//   - tcp://localhost:3000
+//   - http://localhost:3000
+//   - https://localhost:3000
+//   - https-insecure://localhost:3000
+//   - https-insecure://localhost:3000/foo
+func ExpandProxyTargetValue(target string, supportedSchemes []string, defaultScheme string) (string, error) {
```

Comment:
This and the `CheckFunnelAccess`, `NodeCanFunnel`, and `CheckFunnelPort` functions could all be moved elsewhere to a different shared util if anyone has strong opinions. The rest of the funcs in this file are receivers on `ServeConfig` or `ServeConfigView`